addons/hr_work_entry_contract/models/hr_work_entry.py

```diff
@@ -54,6 +54,17 @@ def _init_column(self, column_name):
                 WHERE _hwe.id = ANY(result.entry_ids)
             """)
 
+    def _get_work_entries_data(self, work_entries_dict):
+        data = self.env['hr.work.entry']
+        for key, work_entries in work_entries_dict.items():
+            actual_duration = sum(0 if work_entry.contract_id.time_credit and work_entry.contract_id.time_credit_type_id == work_entry.work_entry_type_id else work_entry.duration for work_entry in work_entries)
+            week_type = "0" if key[0].isocalendar()[1] % 2 == 0 else "1"
+            total_work_entries = key[1].contract_id.resource_calendar_id.attendance_ids.filtered(lambda x: x.week_type == week_type and int(x.dayofweek) == key[0].weekday() and x.day_period != 'lunch')
```

Comment:
You have to fetch the correct contract to get the correct calendar:
```suggestion
    def _get_work_entries_data(self, work_entries_dict):
        data = self.env['hr.work.entry']
        date_from = min(self.mapped('date_start'))
        date_from = max(self.mapped('date_end'))
        contract = self.employee_id._get_contracts(self, date_from, date_to, states=['open', 'close'])
        for key, work_entries in work_entries_dict.items():
            actual_duration = sum(0
                if work_entry.contract_id.time_credit and work_entry.contract_id.time_credit_type_id == work_entry.work_entry_type_id
                else work_entry.duration
                 for work_entry in work_entries)
            week_type = "0" if key[0].isocalendar()[1] % 2 == 0 else "1"
            contract = contracts.filtered(
                lambda c: c.date_start <= work_entry.date_start
                    and not c.date_end or c.date_end >= work_entry.date_end
                    and c.employee_id = key[1]
            )[0]
            total_work_entries = contract.resource_calendar_id.attendance_ids.filtered(lambda x: x.week_type == week_type and int(x.dayofweek) == key[0].weekday() and x.day_period != 'lunch')
```

Reply:
Hello @Bertrand2 
We can simply use the contract_id of the work entry itself
Thank you :innocent: